app/controllers/ansible_repository_controller.rb

```diff
@@ -48,10 +48,20 @@ def button
         page.replace("gtl_div", :partial => "layouts/gtl")
       end
     when "ansible_repository_reload" # repository reload
-      show
-      render :update do |page|
-        page << javascript_prologue
-        page.replace("main_div", :template => "ansible_repository/show")
+      if @display == "output"
+        show
```

Comment:
the show method is repeated on both if and else, so we can take it out

Reply:
maybe we can do something like this - 

```
show
if @display == "output"
  @display = "output" # Remove this line if not needed
  render_update("output_div", "output")
else
  render_update("main_div", "show")
end

def render_update(div_id, partial)
  render :update do |page|
    page << javascript_prologue
    page.replace(div_id, :partial => "ansible_repository/#{partial}")
  end
end
```

can reuse the `render_update` for the case when `pressed = ansible_repositories_reload`